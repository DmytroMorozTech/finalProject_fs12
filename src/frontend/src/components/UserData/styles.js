import {makeStyles} from '@material-ui/core/styles'

export default makeStyles(() => ({

  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  avatarWrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  avatar: {
    width: '40%',
    borderRadius: '50%'
  },
  profileButton: {
    width: '60%',
    marginTop: 15
  },
  userDataWrapper: {
    marginLeft: 15
  },
  userName: {
    fontSize: 14,
    fontWeight: 'bold'
  },
  userProfession: {
    width: '50%',
    fontSize: 12,
    marginBottom: 30
  }
}))