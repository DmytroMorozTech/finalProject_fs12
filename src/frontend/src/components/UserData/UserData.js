import React from 'react'
import styles from './styles'
import Button from '@material-ui/core/Button'
import { Link } from 'react-router-dom'
import { MenuItem } from '@material-ui/core'
import {useDispatch, useSelector} from 'react-redux'
import {activeUserSelector} from '../../redux/User/userSelector'
import Image from '../../shared/Image/Image'
import {signOutAction} from '../../redux/User/userActions'

function UserData () {
  const classes = styles()
  const activeUser = useSelector(activeUserSelector)
  const activeUserId = activeUser.id
  const dispatch = useDispatch()

  const signOut = () => {
    dispatch(signOutAction())
  }

  return (
    <div className={classes.root}>
      <MenuItem component={Link} to={`/profiles/${activeUserId}`} className={classes.menuItem}>
        <div className={classes.avatarWrapper}>
          {/* <img className={classes.avatar} src={activeUser.avatarPublicId} alt={}/> */}
          <Image
            imageUrl={activeUser.avatarPublicId}
            alt={`Avatar of ${activeUser.fullName}`}
            className={classes.avatar}
            type={'smallAvatar'}
          />
          <div className={classes.userDataWrapper}>
            <span className={classes.userName}>{activeUser.fullName}</span>
            <div className={classes.userProfession}>{activeUser.positionAndCompany}</div>
          </div>
        </div>
      </MenuItem>
      <Button component={Link} to={`/profiles/${activeUserId}`} className={classes.profileButton} variant="outlined" color="primary">
        View Profile
      </Button>
      <Button className={classes.profileButton} variant="outlined" color="secondary" onClick={() => signOut()}>
        Sign Out
      </Button>
    </div>
  )
}

export default UserData
