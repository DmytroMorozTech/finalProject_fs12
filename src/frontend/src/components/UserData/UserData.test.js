import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import UserData from './UserData'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<UserData/>)

describe('should render UserData component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let activeUser

  let component

  beforeEach(() => {
    activeUser = {id: 1}

    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(activeUser)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should work post like dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('SharedButton[id=\'SignOut\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
