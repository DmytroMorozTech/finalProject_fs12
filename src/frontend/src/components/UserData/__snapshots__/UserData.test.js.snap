// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserData component to match snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(MenuItem))
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    to="/profiles/1"
  >
    <div>
      <Image
        alt="Avatar of undefined"
        type="smallAvatar"
      />
      <div>
        <span />
        <div />
      </div>
    </div>
  </WithStyles(ForwardRef(MenuItem))>
  <SharedButton
    color="primary"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    to="/profiles/1"
    variant="outlined"
  >
    View Profile
  </SharedButton>
  <SharedButton
    color="secondary"
    id="SignOut"
    onClick={[Function]}
    variant="outlined"
  >
    Sign Out
  </SharedButton>
</div>
`;
